src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -835,39 +834,43 @@ export type SpokePoolFillFilter = SpokePoolEventFilter & {
  */
 export async function getFills(
   chainId: number,
+  hubPoolClient: HubPoolClient,
   spokePoolClients: SpokePoolClients,
   filter: SpokePoolFillFilter = {}
-): Promise<FillWithBlock[]> {
+): Promise<(FillWithBlock & { quoteBlockNumber: number })[]> {
   const spokePoolClient = spokePoolClients[chainId];
   assert(isDefined(spokePoolClient));
 
-  const { originChainId, repaymentChainId, relayer, isSlowRelay, fromBlock } = filter;
+  const { originChainId, repaymentChainId, relayer, isSlowRelay, fromBlock, toBlock } = filter;
 
-  const fills = await filterAsync(spokePoolClient.getFills(), async (fill) => {
-    if (isDefined(fromBlock) && fromBlock > fill.blockNumber) {
-      return false;
-    }
-
-    // @dev tsdx and old Typescript seem to prevent dynamic iteration over the filter, so evaluate the keys manually.
-    if (
-      (isDefined(originChainId) && fill.originChainId !== originChainId) ||
-      (isDefined(repaymentChainId) && fill.repaymentChainId !== repaymentChainId) ||
-      (isDefined(relayer) && fill.relayer !== relayer) ||
-      (isDefined(isSlowRelay) && fill.updatableRelayData.isSlowRelay !== isSlowRelay)
-    ) {
-      return false;
-    }
+  const fills = (
+    await mapAsync(spokePoolClient.getFills(), async (fill) => {
```

Comment:
What's the reason for switching to use `mapAsync` here insteead of `filterAsync` ? As far as I can see, `filterAsync` does what we want and doesn't require the subsequent loop to filter out all the `undefined` instances.

Reply:
Ah - I see, it's because `quoteBlockNumber`  is being added to each fill.

Reply:
yep